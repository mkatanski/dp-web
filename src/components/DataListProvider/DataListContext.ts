import { ReducerName } from "store";
import { createContextWithPartial } from "utils/context/createContextWithPartial";

export type DataListStateContextProps = {
  reducerName: ReducerName;
};

export type DataListActionContextProps = {
  onPaginationChange: (newOffset: number) => void;
};

export const DataListStateContext = createContextWithPartial<
  DataListStateContextProps
>({});

export const DataListActionContext = createContextWithPartial<
  DataListActionContextProps
>({});
