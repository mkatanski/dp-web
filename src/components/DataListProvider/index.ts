export * from "./DataListProvider";
export * from "./DataListContext";
