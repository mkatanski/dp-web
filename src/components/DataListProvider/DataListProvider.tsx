import React from "react";
import { ReducerName } from "store";
import { DataListStateContext, DataListActionContext } from "./DataListContext";

export type DataListProviderProps = React.PropsWithChildren<{
  onPaginationChange?: (newOffset: number) => void;
  reducerName: ReducerName;
}>;

export const DataListProvider: React.FC<DataListProviderProps> = ({
  children,
  reducerName,
  onPaginationChange = () => undefined
}: DataListProviderProps) => {
  return (
    <DataListStateContext.Provider value={{ reducerName }}>
      <DataListActionContext.Provider value={{ onPaginationChange }}>
        {children}
      </DataListActionContext.Provider>
    </DataListStateContext.Provider>
  );
};
