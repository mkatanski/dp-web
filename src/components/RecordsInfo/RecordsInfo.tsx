import React, { useContext } from "react";
import { usePaginationData } from "hooks/usePaginationData";
import { Typography } from "@material-ui/core";
import { DataListStateContext } from "components/DataListProvider";

export type RecordsInfoProps = {};

export const RecordsInfo: React.FC<RecordsInfoProps> = () => {
  const { reducerName } = useContext(DataListStateContext);

  const { limit, offset, totalCount } = usePaginationData(reducerName);

  const page = offset > 0 ? offset / limit : 0;
  const totalPages = Math.ceil(totalCount / limit);

  return (
    <Typography color="textSecondary" gutterBottom variant="body2">
      {totalCount} records found. Page {page + 1} of {totalPages}
    </Typography>
  );
};
