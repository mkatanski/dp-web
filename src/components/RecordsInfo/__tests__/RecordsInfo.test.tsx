import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

import { RecordsInfo } from "../RecordsInfo";
import { DataListStateContext } from "components/DataListProvider";

jest.mock("hooks/usePaginationData", () => ({
  ...require.requireActual("hooks/usePaginationData"),
  usePaginationData: jest
    .fn()
    .mockReturnValue({ limit: 10, offset: 20, totalCount: 100 })
}));

describe("components", () => {
  describe("RecordsInfo", () => {
    it("should render RecordsInfo component", () => {
      const tree = TestRenderer.create(
        <DataListStateContext.Provider
          value={{ reducerName: "deploymentsReducer" }}
        >
          <RecordsInfo />
        </DataListStateContext.Provider>
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
