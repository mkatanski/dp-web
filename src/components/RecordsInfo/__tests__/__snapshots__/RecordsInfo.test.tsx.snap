// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components RecordsInfo should render RecordsInfo component 1`] = `
<div
  data-props={
    Object {
      "color": "textSecondary",
      "gutterBottom": true,
      "variant": "body2",
    }
  }
  id="MUI.Typography"
>
  100
   records found. Page 
  3
   of 
  10
</div>
`;
